lib/Controller/QuickActionsController.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+namespace OCA\Mail\Controller;
+
+use OCA\Mail\AppInfo\Application;
+use OCA\Mail\Exception\ServiceException;
+use OCA\Mail\Http\JsonResponse;
+use OCA\Mail\Http\TrapError;
+use OCA\Mail\Service\QuickActionsService;
+use OCP\AppFramework\Controller;
+use OCP\AppFramework\Db\DoesNotExistException;
+use OCP\AppFramework\Http;
+use OCP\AppFramework\Http\Attribute\NoAdminRequired;
+use OCP\IRequest;
+
+class QuickActionsController extends Controller {
+	private ?string $uid;
+
+	public function __construct(
+		IRequest $request,
+		?string $userId,
+		private QuickActionsService $quickActionsService,
+	) {
+		parent::__construct(Application::APP_ID, $request);
+		$this->uid = $userId;
+	}
+
+	/**
+	 * @NoAdminRequired
+	 *
+	 * @return JsonResponse
+	 */
+	#[TrapError]
+	public function index(): JsonResponse {
+		if ($this->uid === null) {
+			return JsonResponse::error('User not found', Http::STATUS_UNAUTHORIZED);
+		}
+		$actions = $this->quickActionsService->findAll($this->uid);
+
+		return JsonResponse::success($actions);
+	}
+
+	/**
+	 * @NoAdminRequired
+	 * @param string $name
+	 *
+	 * @return JsonResponse
+	 */
+	#[TrapError]
+	public function create(string $name, int $accountId): JsonResponse {
+		if ($this->uid === null) {
+			return JsonResponse::error('User not found', Http::STATUS_UNAUTHORIZED);
+		}
+		try {
+			$quickAction = $this->quickActionsService->create($name, $accountId);
```

Comment:
This allows me to create an action for someone else's account. Please check if the current user owns the account.